src/valid/interface.rs

```diff
@@ -545,6 +545,33 @@ impl super::Validator {
             return Err(EntryPointError::UnexpectedWorkgroupSize.with_span());
         }
 
+        #[cfg(feature = "validate")]
+        if ep.stage == crate::ShaderStage::Vertex {
+            let has_position = match ep.function.result {
```

Comment:
If you check for this just after calling `VaryingContext::validate` on `ep.function.result`, then `crate::BuiltIn::Position { invariant: false }` will be in `result_built_ins` if it was present anywhere on the entry point's return value. You'll have to arrange to check even when `ep.function.result` is `None`, but even so, checking that might be simpler than doing a scan here.

Reply:
Good idea 👍